src/test/java/wooteco/subway/line/LineAcceptanceTest.java

```diff
@@ -0,0 +1,213 @@
+    void createInValidLine() {
+        ExtractableResponse<Response> response = createLineInsertResponse(" ", " ");
+        ErrorResponse errorResponse = response.body().as(ErrorResponse.class);
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+        assertThat(errorResponse.getDetail()).contains("Validation failed for argument [0] in public org.springframework.http.ResponseEntity<wooteco.subway.line.LineResponse> wooteco.subway.line.LineController.createLine(wooteco.subway.line.LineRequest) with 2 errors:");
+        assertThat(errorResponse.getMessage()).isEqualTo("VALIDATION_FAILED");
+    }
+
+    @Test
+    @DisplayName("기존에 존재하는 지하철 노선 이름으로 지하철노선을 생성한다.")
+    void createLineWithDuplicateName() {
+        // given & when
+        createLineInsertResponse("초록색", "2호선");
+        ExtractableResponse<Response> response = createLineInsertResponse("초록색", "2호선");
+        ErrorResponse errorResponse = response.body().as(ErrorResponse.class);
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
+        assertThat(errorResponse.getDetail()).contains("존재하는 노선 이름입니다.");
+        assertThat(errorResponse.getMessage()).contains("LINE_EXCEPTION");
+    }
+
+    @Test
+    @DisplayName("지하철 노선을 조회한다.")
+    void getLines() {
+        /// given
+        ExtractableResponse<Response> createResponse1 = createLineInsertResponse("초록색", "2호선");
+        ExtractableResponse<Response> createResponse2 = createLineInsertResponse("파란색", "1호선");
+
+        // when
+        ExtractableResponse<Response> response = RestAssured.given()
+                .when()
+                .get("/lines")
+                .then()
+                .extract();
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
+        List<Long> expectedLineIds = Stream.of(createResponse1, createResponse2)
+                .map(it -> Long.parseLong(it.header("Location")
+                        .split("/")[2]))
+                .collect(Collectors.toList());
+        List<Long> resultLineIds = response.jsonPath()
+                .getList(".", LineResponse.class)
+                .stream()
+                .map(LineResponse::getId)
+                .collect(Collectors.toList());
+
+        assertThat(resultLineIds).containsAll(expectedLineIds);
+    }
+
+    @Test
+    @DisplayName("지하철 노선 1개를 조회한다.")
+    void getLine() {
+        ExtractableResponse<Response> extract = createLineInsertResponse("초록색", "2호선");
+        String uri = extract.header("Location");
+
+        ExtractableResponse<Response> response = RestAssured.given()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when()
+                .get(uri)
+                .then()
+                .extract();
+
+        assertThat(response.jsonPath().getString("name")).isEqualTo("2호선");
+        assertThat(response.jsonPath().getString("color")).isEqualTo("초록색");
```

Comment:
```suggestion
        LineResponse lineResponse = response.as(LineResponse.class);

        assertThat(lineResponse)
            .usingRecursiveComparison()
            .isEqualTo(new LineResponse(1L, "2호선", "초록색"));
```

이런 방식도 가능해요. 응답 객체의 필드가 많아질수록 특히 유용하겠죠?

Reply:
오.. 완전 좋아요...